command/var_put.go

```diff
@@ -0,0 +1,472 @@
+	}
+
+	if err := flags.Parse(args); err != nil {
+		c.Ui.Error(commandErrorText(c))
+		return 1
+	}
+
+	args = flags.Args()
+
+	// Manage verbose output
+	verbose := func(_ string) {} //no-op
+	if doVerbose {
+		verbose = func(msg string) {
+			c.Ui.Warn(msg)
+		}
+	}
+	// Parse the check-index
+	checkIndex, enforce, err = parseCheckIndex(checkIndexStr)
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Error parsing check-index value %q: %v", checkIndexStr, err))
+		return 1
+	}
+
+	// Pull our fake stdin if needed
+	stdin := (io.Reader)(os.Stdin)
+	if c.testStdin != nil {
+		stdin = c.testStdin
+	}
+
+	switch {
+	case len(args) < 1:
+		c.Ui.Error(fmt.Sprintf("Not enough arguments (expected >1, got %d)", len(args)))
+		c.Ui.Error(commandErrorText(c))
+		return 1
+	case len(args) == 1 && !isArgStdinRef(args[0]) && !isArgFileRef(args[0]):
+		c.Ui.Error("Must supply data")
+		c.Ui.Error(commandErrorText(c))
+		return 1
+	}
+
+	if err = c.validateInputFlag(); err != nil {
+		c.Ui.Error(err.Error())
+		c.Ui.Error(commandErrorText(c))
+		return 1
+	}
+
+	if err := c.validateOutputFlag(); err != nil {
+		c.Ui.Error(err.Error())
+		c.Ui.Error(commandErrorText(c))
+		return 1
+	}
+
+	arg := args[0]
+	switch {
+	// Handle first argument: can be -, @file, «var path»
+	case isArgStdinRef(arg):
+
+		// read the specification into memory from stdin
+		stat, _ := os.Stdin.Stat()
+		if (stat.Mode() & os.ModeCharDevice) == 0 {
+			c.contents, err = io.ReadAll(os.Stdin)
+			if err != nil {
+				c.Ui.Error(fmt.Sprintf("Error reading from stdin: %s", err))
+				return 1
+			}
+		}
+		verbose(fmt.Sprintf("Reading whole %s variable specification from stdin", strings.ToUpper(c.inFmt)))
+
+	case isArgFileRef(arg):
+		// ArgFileRefs start with "@" so we need to peel that off
+		filepath := arg[1:]
+		// detect format based on extension
+		p := strings.Split(filepath, ".")
+		if len(p) < 2 {
+			c.Ui.Error(fmt.Sprintf("Unable to determine format of %s; Use the -in flag to specify it.", filepath))
+			return 1
+		}
+		// detect format based on file extension
+		switch strings.ToLower(p[len(p)-1]) {
+		case "json":
+			c.inFmt = "json"
+		case "hcl":
+			c.inFmt = "hcl"
+		default:
+			c.Ui.Error(fmt.Sprintf("Unable to determine format of %s; Use the -in flag to specify it.", filepath))
+			return 1
+		}
+
+		verbose(fmt.Sprintf("Reading whole %s variable specification from %q", strings.ToUpper(c.inFmt), filepath))
+		c.contents, err = os.ReadFile(filepath)
+		if err != nil {
+			c.Ui.Error(fmt.Sprintf("Error reading %q: %s", filepath, err))
+			return 1
+		}
+	default:
+		path = sanitizePath(arg)
+		verbose(fmt.Sprintf("Writing to path %q", path))
+	}
+
+	args = args[1:]
+	switch {
+	// Handle second argument: can be -, @file, or kv
+	case len(args) == 0:
+		// no-op
+	case isArgStdinRef(args[0]):
+		verbose(fmt.Sprintf("Creating variable %q using specification from stdin", path))
+		stat, _ := os.Stdin.Stat()
+		if (stat.Mode() & os.ModeCharDevice) == 0 {
```

Comment:
Is the idea here that you're preventing someone from accidentally doing something like `cat /dev/urandom | nomad var put`?